Serialize/NPDU.cs

```diff
@@ -118,14 +119,15 @@ public static BacnetNpduControls DecodeFunction(byte[] buffer, int offset)
                 buffer.buffer[buffer.offset++] = (byte)((source.net & 0xFF00) >> 8);
                 buffer.buffer[buffer.offset++] = (byte)((source.net & 0x00FF) >> 0);
                 // Modif FC
-                if (destination.net == 0xFFFF)
+                // DAL this used to encode the destination address, which is clearly wrong...
+                if (source.net == 0xFFFF)
                     buffer.buffer[buffer.offset++] = 0;
                 else
                 {
-                    buffer.buffer[buffer.offset++] = (byte)destination.adr.Length;
-                    if (destination.adr.Length > 0)
+                    buffer.buffer[buffer.offset++] = (byte)source.adr.Length;
+                    if (source.adr.Length > 0)
                     {
-                        foreach (var t in destination.adr)
+                        foreach (var t in source.adr)
                             buffer.buffer[buffer.offset++] = t;
```

Comment:
Good catch, this was already fixed in YABE. However, in YABE it no longer relies on `source.net` and doesn't ignore `source.adr` if the net is 0xFFFF. Which way you think is better? Below YABE version

```csharp
if (has_source)
{
    buffer.buffer[buffer.offset++] =(byte)((source.net & 0xFF00) >> 8);
    buffer.buffer[buffer.offset++] = (byte)((source.net & 0x00FF) >> 0);
    buffer.buffer[buffer.offset++] = (byte)source.adr.Length;
    if (source.adr.Length > 0)
    {
        for (int i = 0; i < source.adr.Length; i++)
            buffer.buffer[buffer.offset++] = source.adr[i];
    }
}
```

Reply:
the spec seems to indicate that the current operation of YABE is correct, so I added a commit to change it.